### DISCUSSION TITLE
How to get soil moisture data as an output?

### DISCUSSION THREAD

**Discussion by PeterBurek** [iiasa/CWatM#5] [Q&A] [answered]
Copied from the old discussion forum:

Luca Guillaumot
15.05.2020, 10:59:21
an CWATM
Hi all,

If you want to get soil moisture data as an output, at the end of CWATM simulation:

- Before running the model: put "OUT_TSS_MonthAvg = sum_w1, sum_w2, sum_w3" at the end of the settings file.
- Results: after your simulation, you obtain three netcdf files in the 'output' directory containing a map for each month of simulation.

CWATM is composed of three soils layers w1, w2 and w3. Water content are in meters. In sum_w1, "sum" refers to the sum weighted by each landcovertype.

These variables are defined in "landcoverType.py" and "soil.py". Also, a list of all variables can be found in "metaNetcdf.xml". Soon, this list will be followed by a variables description and where to find these variables in the code.

----------------

Mikhail..
15.05.2020, 11:59:08

Using MAP will output a map, while TSS will output time series for designated points. 

For example,
OUT_MAP_MonthAvg = sum_w1, sum_w2, sum_w3

totalSto is another possible output totalling the soil moisture of all soil layers, as well as intercept storage, snow cover, and flooded rice paddy water. 
For example,
OUT_MAP_Daily = totalSto

------

**Comment by mikhailsmilovic** [comment, marked as answer]
For soil moisture in soil layers 1, 2, and 3 as a monthly average or daily map:
OUT_MAP_MonthAvg = **sum_w1, sum_w2, sum_w3**
OUT_MAP_Daily = **sum_w1, sum_w2, sum_w3**

There are two other options:
1. **sum_soil** = **sum_w1** + **sum_w2** + **sum_w3** + **sum_topwater**, where **sum_topwater** includes the flood water in paddy rice fields. If one does not have paddy rice fields, sum_soil would be just the sum of the soil layers
2. **totalSto** is the soil moisture of all soil layers, flooded rice paddy water, intercept storage, and snow cover.
